ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java

```diff
@@ -777,7 +790,7 @@ private void addReward(Block block) {
         if (block.getUncleList().size() > 0) {
             for (BlockHeader uncle : block.getUncleList()) {
                 track.addBalance(uncle.getCoinbase(),
-                        new BigDecimal(BLOCK_REWARD).multiply(BigDecimal.valueOf(8 + uncle.getNumber() - block.getNumber()).divide(new BigDecimal(8))).toBigInteger());
+                        new BigDecimal(BLOCK_REWARD).multiply(BigDecimal.valueOf(8 + uncle.getNumber() - block.getNumber()).divide(new BigDecimal(8), RoundingMode.HALF_UP)).toBigInteger());
```

Comment:
Assuming the int-based calculation is implemented as desired, we need to adjust the tests.